pkgs/by-name/si/signal-desktop/generic.nix

```
@@ -101,6 +106,8 @@ stdenv.mkDerivation rec {
     downloadToTemp = true;
     nativeBuildInputs = [
```

I would probably only use the required archive tool for the platform:
```suggestion
   nativeBuildInputs =
      [ asar ]
      ++ (lib.optionals (ARCH == "arm64") [ libarchive ])
      ++ lib.optionals (ARCH == "x64") [ dpkg ];
```